manager.py

```diff
@@ -298,6 +311,8 @@ def sleep_until(self, end_time):
             # this shouldn't happen now that SleepError is raised and handled
             raise RuntimeError
         time.sleep(sleeptime)
+        if self.quit_after_interval and retry:
+            self.sleep_until(end_time, retry=False)
```

Comment:
Add a comment explaining that the SIGQUIT somehow interrupts the `time.sleep`, which is why the retry is needed